tests/core/memory/test_memory.py

```diff
@@ -0,0 +1,75 @@
+import pytest
+
+from evm.vm.memory import (
+    Memory,
+)
+from evm.exceptions import (
+    ValidationError,
+)
+
+
+@pytest.fixture
+def memory():
+    return Memory()
+
+
+@pytest.fixture
+def memory32():
+    memory = Memory()
+    memory.extend(0, 32)
+    return memory
+
+
+@pytest.mark.parametrize("start_position,size,value", ((0, 4, b'1010'),))
+def test_write(memory32, start_position, size, value):
+    memory32.write(start_position, size, value)
+    assert memory32.bytes == value + bytearray(32 - size)
```

Comment:
This is a confusing assertion for me.  It's not easy to understand what exactly is being tested.